src/storage/sql/mod.rs

```diff
@@ -45,13 +45,14 @@ impl SqlStorage {
     }
 
     fn insert_snippet(&self, conn: &PgConnection, snippet: &Snippet) -> Result<i32, StorageError> {
+        let now = chrono::Utc::now();
         let result = diesel::insert_into(snippets::table)
             .values((
                 snippets::slug.eq(&snippet.id),
                 snippets::title.eq(&snippet.title),
                 snippets::syntax.eq(&snippet.syntax),
-                snippets::created_at.eq(diesel::dsl::now),
-                snippets::updated_at.eq(diesel::dsl::now),
+                snippets::created_at.eq(snippet.created_at.unwrap_or(now)),
+                snippets::updated_at.eq(snippet.updated_at.unwrap_or(now)),
```

Comment:
So, this change essentially means now that the current timestamp is generated by the server code, not by the database (as it was before). Do we need this change because of the type requirement set by `.unwrap_or`?

Reply:
> So, this change essentially means now that the current timestamp is generated by the server code, not by the database (as it was before). 

Correct. I did not want to change this, but...

> Do we need this change because of the type requirement set by .unwrap_or?

That's actually a requirement of Rust's type system. `eq()` is a generic function that implements one of `diesel`'s SQL value traits. You can't have a *single* expression in which you conditionally pass different values to `eq()` (`diesel::dsl::now` or `snippet.created_at`), because they have different types, and thus, the compiler needs to generate/call two different versions of `eq()`. So we basically a have a choice between:

* sticking to `DateTime` values, which requires generating the timestamp on the client, and not in the database server
* having a large `if` that would repeat the same `diesel` query expression in both branches with the only difference being in the value which is passed to `eq()`